airflow/providers/databricks/hooks/databricks_sql.py

```diff
@@ -138,15 +141,39 @@ def get_conn(self) -> Connection:
             )
         return self._sql_conn
 
+    @overload
```

Comment:
I’m not quite sure how Mypy handles this; what happens if we omit `overload` signatures in subclasses? Do they get shadowed completely?

Reply:
That's a good question and I have no idea. I can mess around and see what happens.

Some of the method signatures have mutations (some defaults change and some kwargs are added), so I think we'd still need to add all the overloads for those at a minimum.

Reply:
Almost forgot to respond to this.

I tried excluding the overloads in the subclasses and Mypy complained about incompatibility.

That said, I did find that 2 of the `run()` subclass methods were superfluous, so this will help reduce the overall complexity of the code changes I am introducing.